ethexe/contracts/src/IRouter.sol

```diff
@@ -65,6 +66,18 @@ interface IRouter {
     /// @param codeId The code ID of the WASM implementation of the created program.
     event ProgramCreated(address actorId, bytes32 indexed codeId);
 
+    /**
+     * @dev Emitted when a slash request is processed by the middleware.
+     * @param slashData An array of SlashData structures containing details of the slash request.
+     */
+    event RequestSlashCommitmentProcessed(bytes32 indexed commitmentHash, IMiddleware.SlashData[] slashData);
+
+    /**
+     * @dev Emitted when a slash request is executed by the middleware.
+     * @param slashId An array of SlashIdentifier structures containing details of the slash execution.
+     */
+    event ExecuteSlashCommitmentProcessed(bytes32 indexed commitmentHash, IMiddleware.SlashIdentifier[] slashId);
```

Comment:
`indexed` is not needed, because the observer is watching many events at once
```suggestion
    event RequestSlashCommitmentProcessed(bytes32 commitmentHash, IMiddleware.SlashData[] slashData);

    /**
     * @dev Emitted when a slash request is executed by the middleware.
     * @param slashId An array of SlashIdentifier structures containing details of the slash execution.
     */
    event ExecuteSlashCommitmentProcessed(bytes32 commitmentHash, IMiddleware.SlashIdentifier[] slashId);
```